simpleline/base.py

```diff
@@ -19,682 +19,70 @@
 
-        This method does not return until the new screen is closed.
+    def __init__(self):
+        """Do not create instance of this class. Use this class as static.
 
-        :param ui: screen to show
-        :type ui: UIScreen instance
-
-        :param args: optional argument, please see switch_screen for details
-        :type args: anything
+        The `initialize()` method must be called before use.
         """
-        # set the third item to True so new loop gets started
-        self._screens.append((ui, args, self.START_MAINLOOP))
-        self._do_redraw()
-
-    def schedule_screen(self, ui, args=None):
-        """Add screen to the bottom of the stack.
-
-        This is mostly useful at the beginning to prepare the first screen hierarchy to display.
+        super().__init__()
 
-        :param ui: screen to show
-        :type ui: UIScreen instance
-
-        :param args: optional argument, please see switch_screen for details
-        :type args: anything
-        """
-        self._screens.insert(0, (ui, args, self.NOP))
+    @classmethod
+    def initialize(cls, renderer=None, event_loop=None):
+        """Create app instance inside of this class.
 
-    def close_screen(self, scr=None):
-        """Close the currently displayed screen and exit it's main loop if necessary.
+        This method can be called multiple times to reset App settings.
 
-        Next screen from the stack is then displayed.
+        :param renderer: renderer used for rendering screens; if not specified use `simpleline.render.renderer.Renderer`
+        :type renderer: instance of `simpleline.render.renderer.Renderer`
 
-        :param scr: if an UIScreen instance is passed it is checked to be the screen
-                    we are trying to close.
-        :type scr: UIScreen instance
+        :param event_loop: event loop used for asynchronous tasks;
+                           if not specified use `simpleline.event_loop.main_loop.MainLoop`
+        :type event_loop: object based on class `simpleline.event_loop.AbstractEventLoop`
         """
-        oldscr, _oldattr, oldloop = self._screens.pop()
-        if scr is not None:
-            assert oldscr == scr
+        if event_loop is None:
+            event_loop = MainLoop()
+        if renderer is None:
+            renderer = Renderer(event_loop)
 
-        # User can react when screen is closing
-        oldscr.closed()
+        cls.__app = cls.AppPimpl(renderer, event_loop)
 
-        # this cannot happen, if we are closing the window,
-        # the loop must have been running or not be there at all
-        assert oldloop != self.START_MAINLOOP
+    @classmethod
+    def renderer(cls):
+        """Get instance of class responsible for rendering of the screen."""
+        return cls.__app.renderer
 
-        # we are in modal window, end it's loop
-        if oldloop == self.STOP_MAINLOOP:
-            raise ExitMainLoop()
+    @classmethod
+    def event_loop(cls):
```

Comment:
Yeah, the screen based pushing should reduce the need to access the static App class to just starting the main loop in most cases.